src/com/unbounce/dogstatsd/core.clj

```diff
@@ -77,25 +80,74 @@
-;;                             (clojure.core/name status))))
-;;             (.withTags tags)
-;;             (.build))]
-;;     (.recordServiceCheckRun client service-check)))
+(defn event
+  "Records an Event.
+
+  This is a datadog extension, and may not work with other servers.
+
+  See http://docs.datadoghq.com/guides/dogstatsd/#events-1
+
+  Note:
+  :date can either be inst? or msecs
+  :priority can be either :normal or :low
+  "
+  [{:keys [title text timestamp hostname aggregation-key priority source-type-name alert-type]} tags]
+  {:pre [(not (nil? title)) (not (nil? text))]}
+  (let [timestamp (or timestamp -1)
+        event (-> (Event/builder)
+                  (.withTitle title)
+                  (.withText text)
+                  (.withHostname hostname)
+                  (.withAggregationKey aggregation-key)
+                  (.withPriority (if priority
+                                   (Event$Priority/valueOf
+                                    (name priority))
+                                   Event$Priority/NORMAL))
+                  (.withSourceTypeName source-type-name)
+                  (.withAlertType (if alert-type
+                                    (Event$AlertType/valueOf
+                                     (name source-type-name))
+                                    Event$AlertType/INFO))
+                  (.withDate ^long timestamp)
+                  (.build))]
+    (.recordEvent client event (str-array tags))))
+
+(defn service-check
+  "Records a ServiceCheck. This is a datadog extension, and may not work with
+  other servers.
+
+  See https://docs.datadoghq.com/agent/agent_checks/#sending-service-checks
+
+  At minimum, the name and status is required in the payload.
+  "
+  [{:keys [name status hostname message check-run-id timestamp]} tags]
+  {:pre [(not (nil? name)) (not (nil? status))]}
+  (let [service-check
+        (-> (com.timgroup.statsd.ServiceCheck/builder)
+            (.withName name)
+            (.withStatus (case status
+                           :ok       ServiceCheck$Status/OK
+                           :warning  ServiceCheck$Status/WARNING
+                           :critical ServiceCheck$Status/CRITICAL
+                           :unknown  ServiceCheck$Status/UNKNOWN))
+            (.withHostname hostname)
+            (.withMessage message)
+            (.withCheckRunId (or check-run-id 0))
+            (.withTimestamp (or timestamp 0))
+            (.withTags (str-array tags))
+            (.build))]
+    (.recordServiceCheckRun client service-check)))
 
 (defn set-value
   [metric value {:keys [tags]}]
   (.recordSetValue client metric value tags))
+
+(comment
+  (setup!)
+
+  (event {:title "foo" :text "things are bad\nfoo"} nil)
+
+  (service-check {:name "hi" :status :warning} nil)
+  (service-check {:name "hi" :status :ok :timestamp 10 :check-run-id 123 :message "foo" :hostname "blah"} nil)
```

Comment:
move/copy to test ns maybe?

Reply:
I think it's ok to leave `comments` in the source code so you don't have to jump into the test-ns. As for testing, I've yet to think of a good test (other than generative tests). Let's defer that til afterwards though, since this API isn't complicated.